cli/src/main/java/com/devonfw/tools/ide/tool/plugin/PluginBasedCommandlet.java

```diff
@@ -93,41 +95,124 @@ public Path getPluginsInstallationPath() {
   }
 
   @Override
-  protected void postInstall(boolean newlyInstalled) {
+  protected void postInstall(boolean newlyInstalled, ProcessContext pc) {
 
-    super.postInstall(newlyInstalled);
-    if (newlyInstalled) {
-      Path pluginsInstallationPath = getPluginsInstallationPath();
-      FileAccess fileAccess = this.context.getFileAccess();
+    super.postInstall(newlyInstalled, pc);
+    Path pluginsInstallationPath = getPluginsInstallationPath();
+    FileAccess fileAccess = this.context.getFileAccess();
+    if (!Files.exists(retrieveEditionMarkerFilePath(getName()))) {
+      this.context.debug("Matching edition marker file for {} was not found, re-installing plugins", getInstalledEdition());
       fileAccess.delete(pluginsInstallationPath);
-      fileAccess.mkdirs(pluginsInstallationPath);
-      installPlugins(getPlugins().getPlugins());
+      if (this.context.getIdeHome() != null) {
+        List<Path> markerFiles = fileAccess.listChildren(this.context.getIdeHome().resolve(".ide"), Files::isRegularFile);
+        for (Path path : markerFiles) {
+          if (path.getFileName().toString().startsWith("plugin." + getName())) {
+            this.context.debug("Plugin marker file {} got deleted.", path);
+            fileAccess.delete(path);
+          }
+        }
+      }
+
+      createEditionMarkerFile();
+    }
+    fileAccess.mkdirs(pluginsInstallationPath);
+    installPlugins(pc);
+  }
+
+  private void installPlugins(ProcessContext pc) {
+    installPlugins(getPlugins().getPlugins(), pc);
+  }
+
+  /**
+   * Creates a marker file for an installed edition.
+   */
+  public void createEditionMarkerFile() {
+    if (this.context.getPluginsPath() != null) {
+      Path pluginsPath = retrieveEditionMarkerFilePath(getName());
+      FileAccess fileAccess = this.context.getFileAccess();
+      fileAccess.mkdirs(this.context.getPluginsPath().resolve(getName()));
+      fileAccess.touch(pluginsPath);
+    }
+  }
+
+  /**
+   * @param toolName the tool name to search for.
+   * @return Path to the edition marker file.
+   */
+  public Path retrieveEditionMarkerFilePath(String toolName) {
+    if (this.context.getPluginsPath() != null) {
+      return this.context.getPluginsPath().resolve(toolName).resolve("." + getInstalledEdition());
```

Comment:
Did we have a misunderstanding here?
This is not what is described in the story #654.
Surely we change acceptance criteria, but if we do, then we should agree on it and change the story first.
In our teams call, I was referring once that `getInstalledEdition()` is a hack and we should write the installed edition to a simple file like we do with the version in `.ide.software.version` but in `.ide.software.edition`.
Did this maybe cause a misunderstanding and lead to this change? Or is this required for other reasons?
My understanding is the following:
If we check the installation and the resolved version or edition changed, we need to entirely reinstall the according IDE (e.g. IntelliJ) what means:
* "uninstall" the IDE
  * delete all plugin marker files for that IDE
  * delete the plugins folder for that IDE
  * delete the software folder (link) for that IDE
* "reinstall" the IDE again with the new edition and version

Reply:
If this was about the following:
> we IMHO need to track the edition of the IDE. The easiest way would be to write the edition to the file $IDE_HOME/plugins/«ide»/.edition

Then it should be this but it also seems to be used differently:
```suggestion
  public Path retrieveEditionMarkerFilePath(String toolName) {
    if (this.context.getPluginsPath() != null) {
      return this.context.getPluginsPath().resolve(toolName).resolve(".edition");
```

Reply:
Completely removed the edition marker logic.